server/api/v1/stories/index.get.ts

```diff
@@ -0,0 +1,157 @@
+import { stories, components, storyTranslatedSlugs } from "~~/server/database/schema"
+import { requireApiKey } from "~~/server/utils/api-key-auth"
+import { eq, and, inArray, gte, desc, like, type SQL, count } from "drizzle-orm"
+import { z } from "zod"
+import { withPagination, withPaginationDecorator } from "~~/server/utils/pagination"
+
+const querySchema = z.object({
+  ids: z
+    .string()
+    .optional()
+    .transform((val) => (val ? val.split(",").map((id) => id.trim()) : undefined)),
+  componentTypes: z
+    .string()
+    .optional()
+    .transform((val) => (val ? val.split(",").map((type) => type.trim()) : undefined)),
+  parent: z.string().optional(),
+  updatedSince: z
+    .string()
+    .optional()
+    .transform((val) => {
+      if (!val) return undefined
+      // Parse date string - handle both date-only and date-time formats
+      let date: Date
+      if (/^\d{4}-\d{2}-\d{2}$/.test(val)) {
+        // Date-only format (YYYY-MM-DD) - parse as UTC midnight
+        const [year, month, day] = val.split("-").map(Number)
+        date = new Date(Date.UTC(year, month - 1, day, 0, 0, 0, 0))
+      } else {
+        date = new Date(val)
+      }
+      if (isNaN(date.getTime())) {
+        throw new Error("Invalid date format for updatedSince")
+      }
+      return date
+    }),
```

Comment:
```suggestion
  updatedSince: z
    .coerce.date()
    .optional(),
```

How about we do this? This should work. Can you test and if it works change and push this change?